mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -1728,6 +1728,24 @@ LogicalResult CppEmitter::emitType(Location loc, Type type) {
   if (auto tType = dyn_cast<TupleType>(type))
     return emitTupleType(loc, tType.getTypes());
   if (auto oType = dyn_cast<emitc::OpaqueType>(type)) {
+    FailureOr<SmallVector<emitc::OpaqueType::ReplacementItem>> items =
+        oType.parseFormatString();
+    if (failed(items))
+      return failure();
+
+    auto fmtArg = oType.getFmtArgs().begin();
+    for (emitc::OpaqueType::ReplacementItem &item : *items) {
+      if (auto *str = std::get_if<StringRef>(&item)) {
+        os << *str;
+      } else {
+        if (failed(emitType(loc, *fmtArg++))) {
```

Comment:
Would be great if we could also allow integers here, but maybe for later (i.e. to emit `Type<17>` with format string `Type<{}>`